File: golang/lib/execution/test_suite_configurator.go
/*
 * Copyright (c) 2021 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package execution

import "github.com/kurtosis-tech/kurtosis-testsuite-api-lib/golang/lib/testsuite"

// Docs available at https://docs.kurtosistech.com/kurtosis-testsuite-api-lib/lib-documentation
type TestSuiteConfigurator interface {
	// Docs available at https://docs.kurtosistech.com/kurtosis-testsuite-api-lib/lib-documentation
	SetLogLevel(logLevelStr string) error

	// Docs available at https://docs.kurtosistech.com/kurtosis-testsuite-api-lib/lib-documentation
	ParseParamsAndCreateSuite(paramsJsonStr string) (testsuite.TestSuite, error)
}

File: typescript/src/lib/execution/test_suite_configurator.ts
/*
 * Copyright (c) 2021 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

import { TestSuite } from "../testsuite/test_suite"; //TODO
import { Result } from "neverthrow";

// Docs available at https://docs.kurtosistech.com/kurtosis-testsuite-api-lib/lib-documentation
export interface TestSuiteConfigurator {
	// Docs available at https://docs.kurtosistech.com/kurtosis-testsuite-api-lib/lib-documentation
	setLogLevel(logLevelStr: string): Error;

	// Docs available at https://docs.kurtosistech.com/kurtosis-testsuite-api-lib/lib-documentation
	parseParamsAndCreateSuite(paramsJsonStr: string): Result<TestSuite, Error>
}